cli/azd/pkg/project/service.go

```diff
@@ -53,35 +55,53 @@ func (svc *Service) Deploy(
 		defer close(result)
 		defer close(progress)
 
-		log.Printf("packing service %s", svc.Config.Name)
+		var deploymentArtifact string
 
-		progress <- "Preparing packaging"
-		artifact, err := svc.Framework.Package(ctx, progress)
-		if err != nil {
-			result <- &ServiceDeploymentChannelResponse{
-				Error: fmt.Errorf("packaging service %s: %w", svc.Config.Name, err),
+		err := commandHooks.InvokeAction(ctx, "package", func() error {
+			log.Printf("packing service %s", svc.Config.Name)
+			progress <- "Preparing packaging"
+
+			artifact, err := svc.Framework.Package(ctx, progress)
+			if err != nil {
+				result <- &ServiceDeploymentChannelResponse{
+					Error: fmt.Errorf("packaging service %s: %w", svc.Config.Name, err),
+				}
 			}
 
+			deploymentArtifact = artifact
+			return nil
+		})
```

Comment:
Here is an example of the `commandHooks.InvokeAction()` in use.  We do what we were doing before but just wrap our code in the `InvokeAction` block.  This will automatically execute and `pre` and `post` `package` hooks for the current executing service.